### PR TITLE
docs: add java snippets to the examples in guides

### PR DESCRIPTION
**Pull request by yury-s**

### CODE REVIEW

File: docs/src/navigations.md [hunk 1]
```diff
@@ -222,6 +270,14 @@ await Promise.all([
 ]);
 ```
 
+```java
+// Using waitForNavigation with a callback prevents a race condition
+// between clicking and waiting for a navigation.
+page.waitForNavigation(() -> { // Waits for the next navigation
+  page.click("a"); // Triggers a navigation after a timeout
```
Comment (aslushnikov) [inline docs/src/navigations.md:277]: remove "aftter a timeout"

File: docs/src/network.md [hunk 2]
```diff
@@ -112,6 +137,13 @@ const [response] = await Promise.all([
 ]);
 ```
 
+```java
+// Use a glob URL pattern
+Response response =page.waitForResponse("**/api/fetch_data", () -> {
```
Comment (aslushnikov) [inline docs/src/network.md:142]: nit: space between "="

**Review by aslushnikov — APPROVED** [review]

File: docs/src/network.md (hunk 2 above)
Comment (yury-s) [reply docs/src/network.md:142]: Done

File: docs/src/navigations.md (hunk 1 above)
Comment (yury-s) [reply docs/src/navigations.md:277]: it'd make the example useless as waitForNavigation would be redundant. I'll update the snippet separately for all langs

**Pull request merged**